### DISCUSSION TITLE
Quick lookup for needed

### DISCUSSION THREAD

**Discussion by AndreasK79** [magicbug/Cloudlog#936] [Ideas / Feature Requests]
I've been thinking about implementing something like this in Cloudlog now for a while. It is similar to something I've built for myself in another system. Today I thought why not have it popup in a dialog, so that it is accessible from everywhere?

Right now it works with ALT-L (for lookup), and the mockup looks like this:
![image](https://user-images.githubusercontent.com/6977712/110936444-2f971a00-8331-11eb-93ca-3e95767ba5a0.png)

Is this something people would like to have? My thought behind it, is to quickly get an overview of where you have worked/not worked (in this example) a DXCC. Nothing is written in stone about it's function, but these are my thoughts.

**Comment by dg9vh** [comment]
Good idea! Will be useful!

**Comment by magicbug** [comment]
I like this idea, often I use the search box to see if there's a gridsquare that I need

**Comment by AndreasK79** [comment]
Well then. We're getting somewhere:
![image](https://user-images.githubusercontent.com/6977712/111003743-ffc73100-8387-11eb-8d26-eb713cec77ac.png)
Still a long way to go.

**Reply by AndreasK79** [reply]
Or not so long. Getting there:
![image](https://user-images.githubusercontent.com/6977712/111029190-879a5300-83fb-11eb-8af7-2ac75574bf8f.png)

I have a few things to figure out still, but it helped with some sleep. I found it was much easier to implement all the queries than I thought at first. Now I just need to figure out the mode/submode correctly, and getting SAT to work.

**Reply by AndreasK79** [reply]
Problem solved, but now comes the hard part. How to speed this thing up. It takes around 10-11 seconds on my machine...

**Comment by dg9vh** [comment]
Could setting some database indices solve the problem?

**Reply by AndreasK79** [reply]
Good question. I don't think so. I think it's just that things take time with looping through stuff. I need to see if I can do things a faster way.

**Reply by AndreasK79** [reply]
Did some profiling (http://codeigniter.com/userguide3/general/profiling.html)
DATABASE:  cloudlog (Lookup:$db)   QUERIES: 43 (10.8980 seconds)  (Hide).
36 of them comes from what I'm doing. I think I need to rework some logic here.

**Reply by AndreasK79** [reply]
2 seconds and 12 queries :) A bit more testing is needed, but I think it's soon ready for some testing by others!

**Comment by dg9vh** [comment]
Congrats! This sounds amazing!

Kim

**Reply by AndreasK79** [reply]
#937 added. Feel free to test it out.

**Reply by oh1jw** [reply]
This works like a charm. Would it be possible to implement query for WWFF and IOTA qso's?
The WWFF qsos's are logged with SIG (wwff) and SIG-info (ohff-0000) values. And afaik there's quite few logging programs which 
support WWFF natively and even the export is not possible any easy way (but that export is totally another thing - althought an export from quick lookup would be nice addon).

**Reply by AndreasK79** [reply]
@oh1jw IOTA is already implemented. 
![image](https://user-images.githubusercontent.com/6977712/111469878-80d74d00-8727-11eb-8368-6c24862cdea5.png)

Other searches are pretty straight forward to add. I can take a look.

What kind of export from Quick lookup are you after?

**Reply by oh1jw** [reply]
> What kind of export from Quick lookup are you after?
>
The lookup result would be exported to adif (with qso-data), what I see more usefull than csv. 
Maybe someone might see this WWFF more useful in Awards -section, but this quick lookup seems to be 
more quicker than using "Awards". 
And what comes to advanced search - there's no short way to find that, you have to first 
make a "fake" search via Search call sign -field (Yes - I see this search -button a bit confusing) and on the opened page you find the actual search tool. 
This went a little bit offtopic.. SRI!

**Reply by oh1jw** [reply]
Another idea for the user defined searches would be the possibility make perma-links for user defined 
search attributes. Aka you make certain rules for advanced search just once and you can save it to shortcut.
....../index.php/search/filter/**user_defined_shortcut**

**Reply by AndreasK79** [reply]
1. I'll see if I can add WWFF to Quick lookup.
2. Would you like WWFF under awards?
3. I don't think an export like that belongs here, though I don't really know where that belongs. Maybe an advanced export section?
4. As for advanced search.... it is kind of hidden, but once you know it's there, it shouldn't be a problem. If you want changes, please start a new thread for this.

**Reply by oh1jw** [reply]
> 1. I'll see if I can add WWFF to Quick lookup.
>
Thanks again Andreas! 

> 2. Would you like WWFF under awards?
>
That would be good in the long run. 

> 3. I don't think an export like that belongs here, though I don't really know where that belongs. Maybe an advanced export section?
>
Yep - you are right. 

> 4. As for advanced search.... it is kind of hidden, but once you know it's there, it shouldn't be a problem. If you want changes, please start a new thread for this.
>
That's true. I will mature and define the details of my idea better after which i come out with a more precise definition with new thread.

**Reply by AndreasK79** [reply]
@oh1jw WWFF is available in the latest code base. I have everything working for award (with export) locally. Just waiting to see if I should make a new PR, or see if the draft PR is the way to go.
